src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -197,6 +196,9 @@ export class SelectBox extends vsSelectBox {
 	}
 
 	public select(index: number): void {
+		if (this._selectedOption && this._optionsDictionary.get(this._selectedOption) === index) { // Not generating an event if the same value is selected.
```

Comment:
I'd specifically do `this._selectedOption === undefined` - an empty value is still a valid option.